spylib/token.py

```diff
@@ -134,18 +135,30 @@ async def __handle_error(self, debug: str, endpoint: str, response: Response):
         stop=stop_after_attempt(5),
         retry=retry_if_exception(not_our_fault),
     )
-    async def execute_rest(self, goodstatus, debug, endpoint, **kwargs) -> Dict[str, Any]:
+    async def execute_rest(
+        self,
+        request: Request,
+        endpoint: str,
+        json: Optional[Dict[str, Any]] = None,
+        debug: str = "",
+    ) -> Dict[str, Any]:
         while True:
             await self.__await_rest_bucket_refill()
-            kwargs['url'] = f'{self.api_url}{endpoint}'
-            kwargs['headers'] = {'X-Shopify-Access-Token': self.access_token}
 
-            response = await self.client.request(**kwargs)
-            if response.status_code == 429:
+            if not self.access_token:
+                raise ValueError("You have not initialized the token for this store. ")
+
+            response = await self.client.request(
+                method=request.method.value,
+                url=f'{self.api_url}{endpoint}',
+                headers={'X-Shopify-Access-Token': self.access_token},
+                json=json,
+            )
+            if response.status_code == Status.TOO_MANY_REQUESTS:
```

Comment:
I think you need to match the value, right?
```suggestion
            if response.status_code == Status.TOO_MANY_REQUESTS.value:
```

Reply:
You're right, I'm surprised that didn't get caught by any of the tests.